Assets/Scripts/Control&Input/ItemSelectSlot.cs

```diff
@@ -52,55 +60,80 @@ public void SetItems(List<Item> items, Item initialItem)
         {
             items.Insert(0, null);
         }
-
-        // After lots of tweaking, this formula seems to translate size correctly
-        var transformedSpacing = spacing * (itemModelScale / spacing) / 2f;
-
+        int itemCount = items.Count;
+        if (itemCount < 6)
+        {
+            for (int i = 0; i < 6 - itemCount; i++)
+            {
+                items.Insert(itemCount + i, items[Mod((itemCount + i), itemCount)]);
+            }
+        }
 
         // Instantiate all items at offset spacing down from holder
         for (var i = 0; i < items.Count; i++)
         {
-            if (items[i] == null) continue;
-
+            // Item is probably an empty extension slot!
+            if (items[i] == null) 
+            {
+                var missing = Instantiate(missingObject, Vector3.zero, itemHolder.rotation, itemHolder);
+                missing.transform.localPosition = Vector3.zero;
+                augmentModels.Add((missing, missing.transform.localPosition));
+                continue;
+            }
             var rotation = Quaternion.Euler(new Vector3(0, 90, -20));
-
-            var instance = Instantiate(items[i].augment, itemHolder.position + Vector3.down * spacing * i, rotation, itemHolder);
+            var instance = Instantiate(items[i].augment, Vector3.zero, rotation, itemHolder);
             instance.transform.ScaleAndParent(itemModelScale, itemHolder);
-            instance.transform.position = itemHolder.position + Vector3.down * transformedSpacing * i;
+            instance.transform.localPosition = itemSlots[i].transform.position;
 
             Augment.DisableInstance(instance, type);
+            augmentModels.Add((instance, instance.transform.position));
         }
 
         selectedIndex = items.IndexOf(initialItem);
 
         // For extensions, we may not find any available item
         if (selectedIndex < 0) selectedIndex = 0;
 
-        ChangeItemDisplayed(selectedIndex, false);
+        StartCoroutine(WaitAndSetWeapons());
+    }
+
+    private IEnumerator WaitAndSetWeapons()
+    {
+        yield return new WaitForSeconds(0.1f);
+        ChangeItemDisplayed(selectedIndex);
     }
 
-    public void Select()
+    public void Select(out Item item)
     {
         background.color = selectedColor;
+        item = SelectedItem;
     }
 
     public void Deselect()
     {
         background.color = deselectedColor;
     }
 
-    public void Previous()
+    public void Previous(out Item item)
     {
         // +count to avoid negative numbers from modulo (C# is an oddball here)
         ChangeItemDisplayed((selectedIndex - 1 + items.Count) % items.Count);
+        LeanTween.moveLocal(tileHolder, itemSlots[5].transform.localPosition, .2f)
```

Comment:
Um... why is this here and not part of the `for` loop in ChangeItemDisplayed?

Reply:
Because I didn't want to animate it on the initial select / not need to the modulo difference for which direction we should move the slots. I also felt like it was slightly easier to read the code like this compared to potentially sending optional bools into the method for deciding direction and wheter to animate or not (which is my prefference, but I concede it might be non-convential in this case)

Reply:
I'm gonna leave this as is for now, feel free to refactor later if you still want 😅